### PR TITLE
Add design proposal with diagrams for feedback

### PR DESCRIPTION
**Pull request by sigalsax**
This PR adds the design for the project as well as diagrams for visibility and feedback

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,54 @@
...
+
+The following represnts the use case diagram for this project.
+
+![Use Case Service Diagram](./useCaseServiceMapping.png)
+
+### Open questions:
+
+- I provided the Use Case Diagram above but do not feel that those (`Get Container Data` and `Extract Logs/Env`) are use cases as these items depend on Troubleshooting to operate correctly. Are there more use cases besides troubleshooting? 
+
+- In the Troubleshooting class should I use a factory method that chooses between 2 different classes (DAP/Conjur) that implements the same interface (ConjurEnv)?
```
Comment (Tovli) [inline README.md]: You tell me :)
in any case I would start with the HLD first to better understand the system.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,54 @@
...
+
+- What language to write this project in?
```
Comment (Tovli) [inline README.md]: English ;) (A.K.A - what language you feel the project has the most chances to be successful?)

File: README.md [hunk 3]
```diff
@@ -1,2 +1,54 @@
...
+
+
+# Phase 0
+- [x] Reach out to developers, PO/Ms, and SEs to understand the pain points and use this feedback to develop tooling that will feature in the project
+
+- [x] Define an MVP that will provide immediate value and a running backlog
+
+- [x] Create the design and map out call chains and possible user interactions
+
+For the first iteration of this tool, I have decided to pursue the first idea and create a tool that will do the following:
```
[the remark this answers is not in the document]
Comment (sigalsax) [reply README.md]: Added a header. 
I don't think I understand your second point about ENV vars fetching a value. Can you expand?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,54 @@
...
+- Parse and extract Conjur logs based on the type user requests
+
+- A .txt file will be returned to the user with the ENV variables listed at the top and all associated log entries
+
+## Design
+
+
+The following defines the sequence diagram for this project.   
+
+![System Sequence Diagram](./troubleshootSystemSequenceDiagram.png)
```
[the remark this answers is not in the document]
Comment (sigalsax) [reply README.md]: In other words "Get Data from Environment Volatility"?

To your second point, are you referencing the use case diagram or system sequence diagram?

File: README.md (hunk 2 above)
Comment (sigalsax) [reply README.md]: Thinking TS b/c I want to use Angular for the UI. I found this wrapper to help bootstrap some Docker processes https://www.npmjs.com/package/node-docker-api

File: README.md [hunk 5]
```diff
@@ -1,2 +1,54 @@
...
+
+The following represnts the use case diagram for this project.
+
+![Use Case Service Diagram](./useCaseServiceMapping.png)
+
+### Open questions:
+
+- I provided the Use Case Diagram above but do not feel that those (`Get Container Data` and `Extract Logs/Env`) are use cases as these items depend on Troubleshooting to operate correctly. Are there more use cases besides troubleshooting?
```
[the remark this answers is not in the document]
Comment (sigalsax) [reply README.md]: what might be a child use case?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,54 @@
...
+
+- Page with mappings of repository names to their pull commands. Ex: `conjur` → `docker pull registry2.itci.conjur.net/conjur` (for both private and public registries)
+
+- Slackbot where if you use a certain #tag (#documentation), will create salesforce cases for documentation  
+
+
+# Phase 0
+- [x] Reach out to developers, PO/Ms, and SEs to understand the pain points and use this feedback to develop tooling that will feature in the project
+
+- [x] Define an MVP that will provide immediate value and a running backlog
```
[the remark this answers is not in the document]
Comment (sigalsax) [reply README.md]: Moved MVP up to be more clear

File: README.md [hunk 7]
```diff
@@ -1,54 +1,80 @@
...
-- In the Troubleshooting class should I use a factory method that chooses between 2 different classes (DAP/Conjur) that implements the same interface (ConjurEnv)?
+- App will save state so that the user does not have to redefine query with each session
+ 
+### Open questions:
 
-- What language to write this project in?
\ No newline at end of file
+- What language to write this project in? 
+    
+    - Angular/Typescript, Node Docker API: https://www.npmjs.com/package/node-docker-api
```
Comment (Tovli) [inline README.md:80]: You plan this to be UI only project?

Comment (sigalsax) [reply README.md]: no, just added a detail description of the backend

**Pull request merged**